compiler/rustc_target/src/spec/mod.rs

```diff
@@ -899,6 +899,26 @@ supported_targets! {
     ("aarch64_be-unknown-linux-gnu_ilp32", aarch64_be_unknown_linux_gnu_ilp32),
 }
 
+/// List of fields from the `json` target that were not used, so that the user can be warned.
+#[derive(Debug, PartialEq)]
+pub struct TargetWarnings {
+    unused_fields: Vec<String>,
+}
+
+impl TargetWarnings {
+    pub fn empty() -> Self {
+        Self { unused_fields: Vec::new() }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.unused_fields.is_empty()
+    }
+
+    pub fn warning_message(&self) -> String {
+        format!("Some fields from target json were unused: {:?}", self.unused_fields)
```

Comment:
Worth spending an effort to avoid using debug repr of vector here and making it more prose like. (e.g. one of the

```
this, that and another fields in the target.json were not used
the target.json file contains unknown fields: this, that, another
```

or similar.)

Additionally, the warning and error messages should not generally begin with a capital letter because they are already preceded by a `error: ` or `warning: `.

Reply:
Done! Although I'm not sure if we should use `target.json` since that won't actually be the name of the file?